### PR TITLE
[ws_pscan_js] Add Username Scanner

### PR DESCRIPTION
**Pull request by eakirtas**

### CODE REVIEW

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+OPCODE_TEXT = 0x1;
+RISK_INFO 	= 0;
+CONFIDENCE_HIGH = 3;
+
+usersList = null;
+extUserManagment = null;
+
+function scan(helper,msg) {
+
+
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js]: One of these blanks can be removed

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            var context;
+
+            for(i = 0; i < contexts.size(); i++){
+                context = contexts.get(i);
+                contextUsers = extUserManagment.getContextUserAuthManager(context.getIndex()).getUsers();
+                if(contextUsers.size() > 0){
+                    for(j = 0; j < contextUsers.size(); j++ ){
+                        usersList.push(contextUsers.get(j));
+                    }
+
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js]: This blank can be removed

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        var username = user.getName();
+        var usernameHashes = getHashes(username);
+
+        Object.keys(usernameHashes).forEach(function(hashType){
+            var hash = new RegExp(usernameHashes[hashType]);
+            if((matches = message.match(usernameHashes[hashType]))!= null) {
+                matches.forEach(function(evidence){
+                    helper.newAlert()
+                        .setRiskConfidence(RISK_INFO, CONFIDENCE_HIGH)
+                        .setName("Username Hash Found")
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js]: Add "(script)"?

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                        .setDescription(getDiscription(user, hashType))
+                        .setParam(counter++)
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js]: I think this is probably used on some handling in the [Find HTML Comments](https://github.com/zaproxy/community-scripts/blob/27efdf1cb39dee4f9dd71d7055e4b5a6ee231c42/passive/Find%20HTML%20Comments.js#L23-L24) script. However: 1) It's an older script. 2) It uses parameters in order to facilitate different grouping of results.

In the majority of cases passive scanners (scripts or other) should not really have/need a "parameter".

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js (hunk 1 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js]: Fixed in 2de0add5b

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js (hunk 2 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js]: Fixed in 2de0add5b

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js (hunk 3 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js]: Fixed in 2de0add5b

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js (hunk 4 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js]: Fixed in ee4f50813

**Pull request closed without merging**